backends/file/client.go

```diff
@@ -34,33 +37,68 @@ func (c *Client) GetValues(keys []string) (map[string]string, error) {
 		return vars, err
 	}
 
-	nodeWalk(yamlMap, "", vars)
+	if len(keys) == 0 {
+		nodeWalk(yamlMap, "", vars)
+	} else {
+		for _, key := range keys {
+			filteredYamlMap, err := filterByPath(key, yamlMap)
+			if err != nil {
+				return vars, err
+			}
+			nodeWalk(filteredYamlMap, key, vars)
```

Comment:
Why not use `nodeWalk` for all data first, and then filter the values from `vars`?

Reply:
yes, you're right, updated PR